src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,32 @@
 import React from 'react';
+import { Status } from '../../types/Status';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { selectFilter, setQuery, setStatus } from '../../features';
+
+const statusOptions: Status[] = ['all', 'active', 'completed'];
 
 export const TodoFilter: React.FC = () => {
+  const { query, status } = useAppSelector(selectFilter);
+
+  const dispatch = useAppDispatch();
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={(event) => dispatch(setStatus(event.target.value as Status))}
```

Comment:
Ensure that the `event.target.value` is a valid `Status` type before dispatching `setStatus`. Consider adding a type guard or validation to prevent runtime errors.